xcdat/dataset.py

```diff
@@ -321,34 +321,42 @@ def decode_non_cf_time(dataset: xr.Dataset) -> xr.Dataset:
     """
     ds = dataset.copy()
     time = get_axis_coord(ds, "T")
-
-    try:
-        calendar = time.attrs["calendar"]
-    except KeyError:
+    time_attrs = time.attrs
+
+    # NOTE: When opening datasets with `decode_times=False`, the "calendar" and
+    # "units" attributes are stored in `.encoding` (unlike `decode_times=True`
+    # which stores them in `.attrs`). Since xCDAT manually decodes non-CF
+    # compliant time coordinates by first setting `decode_times=False`, the
+    # "calendar" and "units" attrs are popped from the `.attrs` dict and stored
+    # in the `.encoding` dict to mimic xarray's behavior.
+    calendar = time_attrs.pop("calendar", None)
+    units_attr = time_attrs.pop("units", None)
```

Comment:
In `decode_non_cf_time()`, the "calendar" and "units" attributes were being saved in both the `.attrs` and `.encoding` dicts.

However, they should only be in the `.encoding` dict **after decoding time coordinates**. To achieve this behavior, we pop these attrs from the `.attrs` dict and later add them to `.encoding`.